apps/framework-cli/src/framework/core/infrastructure/topic_sync_process.rs

```diff
@@ -57,3 +65,36 @@ impl TopicToTableSyncProcess {
         )
     }
 }
+
+impl TopicToTopicSyncProcess {
+    pub fn from_migration_function(function: &StreamingFunction) -> Self {
+        let source_topic = Topic::from_data_model(&function.source_data_model);
+        let (source_for_func, _) = Topic::from_migration_function(function);
+        TopicToTopicSyncProcess {
+            source_topic_id: source_topic.id(),
+            target_topic_id: source_for_func.id(),
+            source_primitive: PrimitiveSignature {
+                name: function.id(),
+                primitive_type: PrimitiveTypes::Function,
+            },
+        }
+    }
+
+    pub fn id(&self) -> String {
+        format!("{}_{}", self.source_topic_id, self.target_topic_id)
```

Comment:
An example ID is `UserActivity_0_0_UserActivity_0_0_UserActivity_0_1_input`

I don't really like how this looks right now, `UserActivity_0_0` is duplicated, the `_` separation is not clear.

I prefer either the `target_topic_id` (`UserActivity_0_0_UserActivity_0_1_input`) or a different separator (`UserActivity_0_0->UserActivity_0_0_UserActivity_0_1_input`)